DigitalLearningSolutions.Data/Services/CourseContentService.cs

```diff
@@ -80,18 +82,67 @@ GROUP BY
             ).FirstOrDefault();
         }
 
+        public bool DoesProgressExist(int candidateId, int customisationId)
+        {
+            return connection.Query<int>(
+                @"SELECT ProgressId
+                        FROM Progress
+                        WHERE CandidateID = @candidateId
+                          AND CustomisationID = @customisationId
+                          AND SystemRefreshed = 0
+                          AND RemovedDate IS NULL",
+                new { candidateId, customisationId }
+            ).Any();
+        }
+
         public int GetProgressId(int candidateId, int customisationId)
         {
-            // TODO HEEDLS-202: change QueryFirstOrDefault to creating progress record if not found
             return connection.QueryFirstOrDefault<int>(
                 @"SELECT ProgressId
                         FROM Progress
                         WHERE CandidateID = @candidateId
-                          AND CustomisationID = @customisationId",
+                          AND CustomisationID = @customisationId
+                          AND SystemRefreshed = 0
+                          AND RemovedDate IS NULL",
                 new { candidateId, customisationId }
             );
         }
 
+        public void InsertNewProgress(int candidateId, int customisationId, int centreId)

```

Comment:
I wouldn't expect this to be three different public methods. I think there should be one public method to GetProcessId (could be renamed to GetOrCreateProgressId to be clear) which will return an existing one or create a new one 🙂